### PR TITLE
MVP for Podman Machine with AppleHV

### PR DESCRIPTION
**Pull request by baude**
this pr is the first pass at enabling podman machine to use the apple hypervisor. there are still several TODO areas like host networking.  once the decision is handled on what host networking should look like, these TODOs should be fairly quick to resolve.  they also will impact the remove methods.

you must also have vfkit (https://github.com/crc-org/vfkit)

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/applehv/machine.go [hunk 1]
```diff
@@ -18,28 +32,596 @@ func GetVirtualizationProvider() machine.VirtProvider {
...
+	machine.HostUser
+	// ImageConfig describes the bootable image
+	machine.ImageConfig
+	// Mounts is the list of remote filesystems to mount
+	Mounts []machine.Mount
+	// Name of VM
+	Name string
+	/*
+		// NetworkVSock is for the user networking
+		NetworkHVSock machine.HVSockRegistryEntry
```
Comment (mheon) [inline pkg/machine/applehv/machine.go:61]: Why is this commented? Can it be removed?

Comment (baude) [reply pkg/machine/applehv/machine.go:61]: no, holding onto it as part of the networking decisions that need to be made.  you will see that as a theme in the code.

File: pkg/machine/applehv/machine.go [hunk 2]
```diff
@@ -18,28 +32,596 @@ func GetVirtualizationProvider() machine.VirtProvider {
...
+	gvproxy, err := cfg.FindHelperBinary("gvproxy", false)
+	if err != nil {
+		return "", 0, err
+	}
+
+	attr.Files = []*os.File{dnr, dnw, dnw}
+	cmd := []string{gvproxy}
+	// Add the ssh port
+	cmd = append(cmd, []string{"-ssh-port", fmt.Sprintf("%d", m.Port)}...)
+	//cmd = append(cmd, []string{"-listen", fmt.Sprintf("vsock://%s", m.NetworkHVSock.KeyName)}...)
```
Comment (ashley-cui) [inline pkg/machine/applehv/machine.go:553]: Commented code, either tag with TODO or remove

File: pkg/machine/errors.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import "errors"
+
+var (
+	ErrNoSuchVM         = errors.New("VM does not exist")
+	ErrWrongState       = errors.New("VM in wrong state to perform action")
+	ErrVMAlreadyExists  = errors.New("VM already exists")
+	ErrVMAlreadyRunning = errors.New("VM already running or starting")
+	ErrMultipleActiveVM = errors.New("only one VM can be active at a time")
+	ErrNotImplemented   = errors.New("functionality not implemented")
+	ForwarderBinaryName = "gvproxy"
```
Comment (rhatdan) [inline pkg/machine/errors.go:12]: Nit: not really an error?

File: pkg/machine/applehv/config.go [hunk 4]
```diff
@@ -1,22 +1,52 @@
-//go:build arm64 && darwin
-// +build arm64,darwin
```
Comment (ashley-cui) [inline pkg/machine/applehv/config.go:2]: Shouldn't these stay? Since apple HV is m1 only?

File: pkg/machine/applehv/rest.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	b, err := json.Marshal(define.VMState{State: string(newState)})
+	if err != nil {
+		return err
+	}
+	payload := bytes.NewReader(b)
+	_, err = vf.post(vf.Endpoint+state, payload)
+	if err != nil {
+		return err
+	}
+	return err
```
Comment (ashley-cui) [inline pkg/machine/applehv/rest.go:89]:
```suggestion
	_, err = vf.post(vf.Endpoint+state, payload)
	return err
```

**Pull request merged**